cylc/flow/suite_events.py

```diff
@@ -51,16 +51,28 @@ def __init__(self, proc_pool):
     @staticmethod
     def get_events_conf(config, key, default=None):
         """Return a named [cylc][[events]] configuration."""
-        for getter in [
+        # Mail doesn't have any defaults in suite.py
+        if 'mail' in config.cfg['cylc']:
+            getters = [
                 config.cfg['cylc']['events'],
-                glbl_cfg().get(['cylc', 'events'])]:
-            try:
+                config.cfg['cylc']['mail'],
+                glbl_cfg().get(['cylc', 'events']),
+                glbl_cfg().get(['cylc', 'mail'])
+            ]
+        else:
+            getters = [
+                config.cfg['cylc']['events'],
+                glbl_cfg().get(['cylc', 'events']),
+                glbl_cfg().get(['cylc', 'mail'])
+            ]
+        value = None
+        for getter in getters:
+            if key in getter:
                 value = getter[key]
-            except KeyError:
-                pass
-            else:
-                if value is not None:
-                    return value
+            elif key.replace('mail ', '') in getter:
```

Comment:
This method is used by both scheduler and task events. Task events are still in the form 'mail *'.

Reply:
Does it make sense to keep that when the `[cylc]`/`[scheduler]` events has moved the mail stuff into its own section? Why be inconsistent?

Reply:
@hjoliver @oliver-sanders - Anyone remember the reasoning behind this?

Reply:
Do you mean (in the flow config case) the apparent discrepancy between this:
```python
SPEC = {
    'scheduler': {
         'events': { 
             'mail events': [VDR.V_STRING_LIST],
        },
        'mail': {
             'from': [VDR.V_STRING],
             'smtp': [VDR.V_STRING],
             'to': [VDR.V_STRING],
             'footer': [VDR.V_STRING],
             'task event interval': [VDR.V_INTERVAL, DurationFloat(300)],
        },
```
and this
```python
    'runtime': {
        '__MANY__': {
               'events': {
                     'mail events': [VDR.V_STRING_LIST, None],
                     'mail from': [VDR.V_STRING],
                     'mail to': [VDR.V_STRING],
                },
```

I guess we decided we need to retain support different tasks to send mail to different addresses.  But not sure why 'mail from` and `mail to` aren't in a separate`mail` section (and with the "mail " prefix removed) for consistency with the `scheduler` section.  This might be a question for @dpmatthews

Reply:
I agree the separate `mail` section would make more sense.